timdex_dataset_api/metadata.py

```diff
@@ -453,3 +472,41 @@ def write_append_delta_duckdb(self, filepath: str) -> None:
         logger.debug(
             f"Append delta written: {output_path}, {time.perf_counter()-start_time}s"
         )
+
+    def build_meta_query(
+        self, table: str, where: str | None, **filters: Unpack["DatasetFilters"]
+    ) -> str:
+        """Build SQL query using SQLAlchemy against metadata schema tables and views."""
```

Comment:
This is admiteddly pretty dense, but it's an operation that is common to many applications where you're translating key/value style arguments into a SQL-like dialect.

The price of those key/value filters -- which _will_ likely be the most common way the dataset is filtered -- is the complexity here of converting that into a SQL `WHERE` clause.  

While SQLAlchemy reflection + query building may seem at a glance more complicated then building a `WHERE` query string, it quickly becomes apparent that's not the case as you start getting into escaping, logic for every possible SQL operator, etc.  

This is a good method to spend some time reading and thinking about, as it's also pretty key for this new approach.